libkbfs/rekey_fsm.go

```diff
@@ -435,25 +456,23 @@ func (m *rekeyFSM) loop() {
 
 // Event implements RekeyFSM interface for rekeyFSM.
 func (m *rekeyFSM) Event(event RekeyEvent) {
-	select {
-	case m.reqs <- event:
-		return
-	default:
-	}
-	go func() {
-		// We use a spinning loop here to avoid doing a naked `m.reqs <- event`,
-		// which in racy conditions could cause a panic due to sending to closed
-		// channel. The spinning loop is fine here since reactToEvent
+	for {
+		// We use a spinning loop here to avoid doing a naked `m.reqs <-
+		// event`, which in racy conditions could cause a panic due to sending
+		// to closed channel. The spinning loop is fine here since reactToEvent
 		// implementations of states are supposed to consume events from m.reqs
 		// pretty fast.
-		for atomic.LoadInt32(&m.reqsClosed) == 0 {
-			select {
-			case m.reqs <- event:
-				return
-			default:
-			}
+		m.muReqs.RLock()
```

Comment:
```go
for {
    func() {
        m.reqsMtx.RLock()
        defer m.reqsMtx.RUnlock()
        ...
    }()
}
```
Otherwise it won't get unlocked when it should.